src/interpreter/runtime/values/module.js

```diff
@@ -437,6 +466,8 @@ export function createInstance(
 
   instantiateInternals(n, allocator, instantiatedInternals, moduleInstance);
 
+  instantiateDataSections(n, allocator, moduleInstance);
```

Comment:
`instantiateDataSections` should be in internal IMO, because it's instantiating the module's internal. It's probably better so because it will hurt the readability, but also adds a additional traversal. What do you think?

Reply:
The reason I added this additional traversal is because [the spec mentions that](https://webassembly.github.io/spec/core/text/modules.html#text-module):

> A module consists of a sequence of fields that can occur in any order.

Therefore, a data section can appear before the memory or global it refers to.

The logic could be moved to `instantiateInternals`, but would require some form of post-traversal processing.

Reply:
Ah that's perfect then! thanks for the clarification